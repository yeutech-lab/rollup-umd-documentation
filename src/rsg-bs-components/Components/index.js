export { default } from 'rsg-components/Components/ComponentsRenderer';
