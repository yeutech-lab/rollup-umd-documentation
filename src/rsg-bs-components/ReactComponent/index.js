export { default } from 'rsg-components/ReactComponent/ReactComponentRenderer';
