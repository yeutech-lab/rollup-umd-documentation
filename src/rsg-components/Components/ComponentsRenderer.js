import React from 'react';
import PropTypes from 'prop-types';

export default function ComponentsRenderer({ children }) {
	return <div>{children}</div>;
}
ComponentsRenderer.propTypes = {
	children: PropTypes.node.isRequired,
};
