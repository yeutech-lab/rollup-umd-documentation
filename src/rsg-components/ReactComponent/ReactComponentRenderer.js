import React from 'react';
import PropTypes from 'prop-types';
import Header from 'bootstrap-styled/lib/Header';
import omit from 'lodash.omit';
import styled from 'styled-components';
import mapToCssModules from 'map-to-css-modules/lib';
import cn from 'classnames';
import Pathline from '../Pathline';

export const defaultProps = {
  theme: {
    styleguide: {
      '$rsg-react-component-margin': '0 0 40px 0',
      '$rsg-react-component-header-margin': '0 0 16px 0',
      '$rsg-react-component-docs-color': '#333',
      '$rsg-react-component-docs-font-size': '16px',
      '$rsg-react-component-tabs-margin': '0 0 16px 0',
      '$rsg-react-component-tabs-button-margin': '0 0 8px 0',
    },
  },
};
/* eslint-disable react/require-default-props */
export const propTypes = {
  /**
   * @ignore
   */
  className: PropTypes.string, // eslint-disable-line react/require-default-props
  /** Name used in id. */
  name: PropTypes.string.isRequired,
  /** Heading element to be rendered. */
  heading: PropTypes.node.isRequired,
  /** Component directory file path. */
  filepath: PropTypes.string,
  /** Component directory path line. */
  pathLine: PropTypes.string,
  /** Tab buttons elements to be rendered. */
  tabButtons: PropTypes.node,
  /** Tab body element to be rendered. */
  tabBody: PropTypes.node,
  /** Description element to be rendered. */
  description: PropTypes.node,
  /** Documentation element to be rendered. */
  docs: PropTypes.node,
  /** Examples elements to be rendered. */
  examples: PropTypes.node,
  /** Toggle isolated style. */
  isolated: PropTypes.bool,
  /** Theme variables. Can be: */
  theme: PropTypes.shape({
    styleguide: PropTypes.shape({
      '$rsg-react-component-margin': PropTypes.string,
      '$rsg-react-component-header-margin': PropTypes.string,
      '$rsg-react-component-docs-color': PropTypes.string,
      '$rsg-react-component-docs-font-size': PropTypes.string,
      '$rsg-react-component-tabs-margin': PropTypes.string,
      '$rsg-react-component-tabs-button-margin': PropTypes.string,
    }),
  }),
  /**
   * Replace or remove a className from the component.
   * See example <a href="https://www.npmjs.com/package/map-to-css-modules" target="_blank">here</a>.
   */
  cssModule: PropTypes.object, // eslint-disable-line react/require-default-props
};
/* eslint-enable react/require-default-props */


const ReactComponentRendererUnstyled = (props) => {
  const {
    className,
    name,
    heading,
    pathLine,
    description,
    docs,
    examples,
    tabButtons,
    tabBody,
    cssModule,
    ...attributes
  } = omit(props, ['theme']);

  return (
    <div
      id={`${name}-container`}
      className={mapToCssModules(cn(className, 'rsg-react-component'), cssModule)}
      {...attributes}
    >
      <Header className="rsg-react-component-header">
        {heading}
        {pathLine && <Pathline>{pathLine}</Pathline>}
      </Header>
      {(description || docs) && (
        <div className="rsg-react-component-docs">
          {description}
          {docs}
        </div>
      )}
      {tabButtons && (
        <div className="rsg-react-component-tabs">
          <div className="rsg-react-component-tabs-button">{tabButtons}</div>
          {tabBody}
        </div>
      )}
      {examples}
    </div>
  );
};

ReactComponentRendererUnstyled.defaultProps = defaultProps;
ReactComponentRendererUnstyled.propTypes = propTypes;

const ReactComponentRenderer = styled(ReactComponentRendererUnstyled)` 
  ${(props) => `
    &.rsg-react-component {
      margin: ${props.theme.styleguide['$rsg-react-component-margin']};
      .rsg-react-component-header {
        margin: ${props.theme.styleguide['$rsg-react-component-header-margin']};
      }
      .rsg-react-component-docs {
        color: ${props.theme.styleguide['$rsg-react-component-docs-color']};
        font-size: ${props.theme.styleguide['$rsg-react-component-docs-font-size']};
      }
      .rsg-react-component-tabs {
        margin: ${props.theme.styleguide['$rsg-react-component-tabs-margin']};
        .rsg-react-component-tab-buttons {
          margin: ${props.theme.styleguide['$rsg-react-component-tabs-button-margin']};
        }
      }
    }
 `}
`;

ReactComponentRenderer.defaultProps = defaultProps;
ReactComponentRenderer.propTypes = propTypes;

export default ReactComponentRenderer;
