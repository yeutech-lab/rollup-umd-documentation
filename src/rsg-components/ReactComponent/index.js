export { default } from 'rsg-components/ReactComponent/ReactComponent';
