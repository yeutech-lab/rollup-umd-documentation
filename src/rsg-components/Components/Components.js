import React from 'react';
import PropTypes from 'prop-types';
import ReactComponent from 'rsg-components/ReactComponent';
import ComponentsRenderer from 'rsg-components/Components/ComponentsRenderer';

export default function Components({ components, depth }) {
	return (
		<ComponentsRenderer>
			{components.map(component => (
				<ReactComponent key={component.filepath} component={component} depth={depth} />
			))}
		</ComponentsRenderer>
	);
}

Components.propTypes = {
	components: PropTypes.array.isRequired,
	depth: PropTypes.number.isRequired,
};
