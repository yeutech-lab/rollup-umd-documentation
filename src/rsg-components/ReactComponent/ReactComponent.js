import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Examples from 'rsg-components/Examples';
import SectionHeading from 'rsg-components/SectionHeading';
import JsDoc from 'rsg-components/JsDoc';
import Markdown from 'rsg-components/Markdown';
import Slot from 'rsg-components/Slot';
import ReactComponentRenderer from 'rsg-components/ReactComponent/ReactComponentRenderer';
import { DOCS_TAB_USAGE } from '../slots';
import { DisplayModes } from '../../consts';

const ExamplePlaceholder =
	process.env.STYLEGUIDIST_ENV !== 'production'
		? require('rsg-components/ExamplePlaceholder').default
		: () => <div />;

export default class ReactComponent extends Component {
	static propTypes = {
		component: PropTypes.object.isRequired,
		depth: PropTypes.number.isRequired,
	};
	static contextTypes = {
		config: PropTypes.object.isRequired,
		displayMode: PropTypes.string,
	};

	constructor(props, context) {
		super(props, context);
		const { showUsage } = context.config;

		this.handleTabChange = this.handleTabChange.bind(this);

		this.state = {
			activeTab: showUsage ? DOCS_TAB_USAGE : undefined,
		};
	}

	handleTabChange(name) {
		this.setState(state => ({
			activeTab: state.activeTab !== name ? name : undefined,
		}));
	}

	render() {
		const { activeTab } = this.state;
		const { displayMode } = this.context;
		const { component, depth } = this.props;
		const { name, slug, filepath, pathLine } = component;
		const { description, examples = [], tags = {} } = component.props;
		if (!name) {
			return null;
		}

		return (
			<ReactComponentRenderer
				name={name}
				slug={slug}
				filepath={filepath}
				pathLine={pathLine}
				docs={<JsDoc {...tags} />}
				description={description && <Markdown text={description} />}
				heading={
					<SectionHeading
						id={slug}
						deprecated={!!tags.deprecated}
						slotName="componentToolbar"
						slotProps={{
							...component,
							isolated: displayMode !== DisplayModes.all,
						}}
						depth={depth}
					>
						{name}
					</SectionHeading>
				}
				examples={
					examples.length > 0 ? (
						<Examples examples={examples} name={name} />
					) : (
						<ExamplePlaceholder name={name} />
					)
				}
				tabButtons={
					<Slot
						name="docsTabButtons"
						active={activeTab}
						props={{ ...component, onClick: this.handleTabChange }}
					/>
				}
				tabBody={<Slot name="docsTabs" active={activeTab} onlyActive props={component} />}
			/>
		);
	}
}
