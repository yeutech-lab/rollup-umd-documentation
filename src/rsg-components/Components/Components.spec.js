import React from 'react';
import ReactComponent from '../ReactComponent';
import Components from './Components';
import ComponentsRenderer from './ComponentsRenderer';

const components = [
	{
		name: 'Foo',
		pathLine: 'components/foo.js',
		filepath: 'components/foo.js',
		props: {
			description: 'Foo foo',
		},
	},
	{
		name: 'Bar',
		pathLine: 'components/bar.js',
		filepath: 'components/bar.js',
		props: {
			description: 'Bar bar',
		},
	},
];

it('should render components list', () => {
	const actual = shallow(<Components components={components} depth={3} />);

	expect(actual).toMatchSnapshot();
});

it('renderer should render components list layout', () => {
	const actual = shallow(
		<ComponentsRenderer>
			<ReactComponent key={0} component={components[0]} depth={3} />
			<ReactComponent key={1} component={components[1]} depth={3} />
		</ComponentsRenderer>
	);

	expect(actual).toMatchSnapshot();
});
