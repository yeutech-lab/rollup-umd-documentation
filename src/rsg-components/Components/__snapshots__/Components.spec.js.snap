// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render components list layout 1`] = `
<div>
  <ReactComponent
    component={
      Object {
        "filepath": "components/foo.js",
        "name": "Foo",
        "pathLine": "components/foo.js",
        "props": Object {
          "description": "Foo foo",
        },
      }
    }
    depth={3}
    key="0"
  />
  <ReactComponent
    component={
      Object {
        "filepath": "components/bar.js",
        "name": "Bar",
        "pathLine": "components/bar.js",
        "props": Object {
          "description": "Bar bar",
        },
      }
    }
    depth={3}
    key="1"
  />
</div>
`;

exports[`should render components list 1`] = `
<ComponentsRenderer>
  <ReactComponent
    component={
      Object {
        "filepath": "components/foo.js",
        "name": "Foo",
        "pathLine": "components/foo.js",
        "props": Object {
          "description": "Foo foo",
        },
      }
    }
    depth={3}
    key="components/foo.js"
  />
  <ReactComponent
    component={
      Object {
        "filepath": "components/bar.js",
        "name": "Bar",
        "pathLine": "components/bar.js",
        "props": Object {
          "description": "Bar bar",
        },
      }
    }
    depth={3}
    key="components/bar.js"
  />
</ComponentsRenderer>
`;
